packages/amplify-environment-parameters/src/resource-parameter-manager.ts

```diff
@@ -13,6 +13,9 @@ export class ResourceParameterManager {
 
   // eslint-disable-next-line jsdoc/require-jsdoc
   setParam(name: string, value: string): void {
+    if (value === undefined) {
+      return;
+    }
```

Comment:
I think since deleteParam is a separate method and TS should prevent passing in undefined values, this behavior is more appropriate

Reply:
in that case I suggest we validate and throw to encourage correct usage.

Reply:
I'm hesitant to throw because we have cases today where we have objects with undefined values and when we serialize them to disk we expect the undefined values to be ignored

Reply:
if that the case, then unsetting would be desired behavior.
manager.set("foo", "bar");
manager.set("foo", null);
should write null or throw.

Reply:
@jhockett do you have opinion on above ?

Reply:
I see what you mean. Yeah I'll change it to remove

Reply:
What if the function name was changed to better reflect that `undefined` would be skipped?
Something like:
```ts
  defineParamString(name: string, value: string): void {
    if (value === undefined) {
      return;
    }
```

Reply:
I'd argue that doesn't really clarify it as you could be "defining" the value to be undefined. It's also unconventional if you are looking for a setter on the object. I'm hoping that the JSDoc + having the type be string is enough to define the intent

Reply:
There's always this discussion when `set` and `remove` apis are present at the same interface.

I think that practical choice is to have set("foo", null/undefined) work same way as remove("foo"). This makes callers write less logic if they want to batch process properties in loops.